partiql-logical/src/lib.rs

```diff
@@ -86,8 +86,11 @@ where
         self.edges.push((src, dst, 0));
     }
 
-    /// Adds a data flow with a branch number. One application of branch number is in data flows to
-    /// [`Join`] operator to distinguish between LHS and RHS of a JOIN.
+    /// Adds a data flow with a branch number.
+    /// TODO: decide if `branch_num` is necessary within the current implementation. JOINs were
```

Comment:
Since I refactored the `EvalJoin` node to not require a left and right `input`, I'm not too sure if `branch_num`s are useful currently. They're probably useful for some other use case we'll have in the future. Updated the documentation to make that point clearer.

Reply:
I'm also generally inclined to remove it and add back later if it's needed. I'll cut a separate issue (https://github.com/partiql/partiql-lang-rust/issues/237) to track and possibly remove so this PR's scope isn't increased too much.

Reply:
Great, would be great to have it before release!